src/app/pages/Swap/form/hooks.ts

```diff
@@ -0,0 +1,112 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+
+import { Token } from '@lifi/sdk';
+import { useDispatch } from 'react-redux';
+
+import {
+  putLifiEvmTokensMetadataAction,
+  putLifiEvmTokensMetadataLoadingAction
+} from 'app/store/evm/swap-lifi-metadata/actions';
+import { TokenSlugTokenMetadataRecord } from 'app/store/evm/swap-lifi-metadata/state';
+import { getEvmSwapTokensMetadata, TokensByChain } from 'lib/apis/temple/endpoints/evm';
+import { ChainID, LifiSupportedChains } from 'lib/apis/temple/endpoints/evm/api.interfaces';
+import { useEvmAccountTokens } from 'lib/assets/hooks/tokens';
+import { toTokenSlug } from 'lib/assets/utils';
+import { EVM_ZERO_ADDRESS } from 'lib/constants';
+import { EvmAssetStandard } from 'lib/evm/types';
+import { LIFI_METADATA_SYNC_INTERVAL } from 'lib/fixed-times';
+import { useEnabledEvmChains } from 'temple/front';
+
+/**
+ * Hook to fetch and provide available EVM tokens for swap operations
+ */
+export const useLifiEvmTokensSlugs = (publicKeyHash: HexString) => {
+  const dispatch = useDispatch();
+
+  const [lifiEvmTokensByChain, setLifiEvmTokensByChain] = useState<TokensByChain>({});
+  const lastFetchTimeRef = useRef<number>(0);
```

Comment:
you should save timestamp to persisted store, otherwise there is no need in this logic, cause tokens will be reloaded every time user refreshes the page

Reply:
yeah, as you wish, you can remove some logic here and add memoization on the backend or you can go with the current approach and save timestamps to` src/app/store/evm/swap-lifi-metadata/state.ts`, but make sure this timestamp is persisted.